gallery/experiments/experimental_abinitio_pipeline_10073_jsb.py

```diff
@@ -0,0 +1,167 @@
+src = RelionSource(
+    starfile_in, pixel_size=pixel_size, max_rows=n_imgs, data_folder=data_folder
+)
+
+# Use ``phase_flip`` to attempt correcting for CTF.
+logger.info("Perform phase flip to input images.")
+src = src.phase_flip().cache()
+
+# Downsample the images.
+logger.info(f"Set the resolution to {img_size} X {img_size}")
+src = src.downsample(img_size).cache()
+
+# Normalize the background of the images.
+src = src.normalize_background().cache()
+
+# Estimate the noise and whiten based on the estimated noise.
+src = src.legacy_whiten().cache()
+
+# Optionally invert image contrast.
+logger.info("Invert the global density contrast")
+src = src.invert_contrast().cache()
+
+# Save the preprocessed image stack.
+src.save(preprocessed_fn, save_mode="single", overwrite=True)
+
+
+# %%
+# Class Averaging
+# ----------------------
+#
+# Now perform classification and averaging for each class.
+
+logger.info("Begin Class Averaging")
+
+avgs = LegacyClassAvgSource(src, n_nbor=n_nbor).cache()
+
+# Save the entire set of class averages to disk so they can be reused.
+avgs.save(class_avg_fn, save_mode="single", overwrite=True)
+
+# We'll continue our pipeline by selecting``n_classes`` from ``avgs``.
+# To capture a broader range of viewing angles, uniformly select every ``k`` image.
+k = (avgs.n - 1) // n_classes
+avgs = avgs[::k].cache()
+
+
+# %%
+# Common Line Estimation
+# ----------------------
+#
+# Estimate orientation of projections and assign to source by
+# applying ``OrientedSource`` to the class averages from the prior
+# step. By default this applies the Common Line with Synchronization
+# Voting ``CLSync3N`` method.  Here additional weighting techniques
+# are applied for common lines detection by customizing the
+# orientation estimator component.
+
+logger.info("Apply custom mask")
+# 10073 benefits from a masking procedure that is more aggressive than the default.
+# Note, since we've manually masked, the default masking is disabled below in `CLSync3N`.
+# This also upcasts to double precision, which is helpful for this reconstruction.
+mask = fuzzy_mask((img_size, img_size), np.float64, r0=0.4 * img_size, risetime=2)
+avgs = ArrayImageSource(avgs.images[:] * mask)
+
+logger.info("Begin Orientation Estimation")
+# Configure the `CLSync3N` algorithm,
+#  customized by enabling weighting and disabling default mask.
+ori_est = CLSync3N(avgs, mask=False, S_weighting=True, J_weighting=True)
+# Handles calling code to find and assign orientations and shifts.
+oriented_src = OrientedSource(avgs, ori_est)
+
+# Save off the set of class average images, along with the estimated orientations and shifts.
+oriented_src.save(oriented_fn, save_mode="single", overwrite=True)
+
+
+# %%
+# Volume Reconstruction
+# ----------------------
+#
+# Using the oriented source, attempt to reconstruct a volume.
+
+logger.info("Begin Volume reconstruction")
+
+# Setup an estimator to perform the back-projection.
```

Comment:
Setup → set up, back-projection → backprojection

Reply:
Changed in both 10028 and 73.